### PR TITLE
client: introduce WithDisableServiceConfig DialOption

### LINKED ISSUE #1998: Support turning off TXT lookups in DNS resolver
Please answer these questions before submitting your issue.

### What version of gRPC are you using?
1.10.0
### What version of Go are you using (`go version`)?
1.10
### What operating system (Linux, Windows, …) and version?
Linux
### What did you do?

Use the DNS resolver
### What did you expect to see?
No log messages
### What did you see instead?
Lots and lots of warning messages

It would be nice if it was possible to configure the resolver to not try to get TXT records to configure grpc.  We don't use that functionality and our logs are filled with warning lines about not being able to resolve or parse TXT records.

### PR DESCRIPTION
**Pull request by lyuxuan**
fix #1998

- also change not found SRV record and TXT record log level from warning to info

### CODE REVIEW

**Comment by lyuxuan** [comment]
fix #1998

File: resolver/resolver.go [hunk 1]
```diff
@@ -86,6 +86,7 @@ type Address struct {
 // BuildOption includes additional information for the builder to create
 // the resolver.
 type BuildOption struct {
+	ServiceConfigOff bool
```
Comment (dfawley) [inline resolver/resolver.go]: Please add a comment.

File: clientconn.go [hunk 2]
```diff
@@ -412,6 +413,14 @@ func WithChannelzParentID(id int64) DialOption {
 	}
 }
 
+// WithServiceConfigOff returns a DialOption that specifies whether resolver
+// should fetch service config.
+func WithServiceConfigOff(a bool) DialOption {
```
Comment (dfawley) [inline clientconn.go]: What if we made this `WithResolverBuilderOptions(opts resolver.BuilderOption)` instead, to be more extensible?

@menghanl

Comment (menghanl) [reply clientconn.go]: But what if we later want to add another field to `resolver.BuilderOption` that's not to be configured by the user?

**Comment by dfawley** [comment]
> But what if we later want to add another field to `resolver.BuilderOption` that's not to be configured by the user?

I can't come up with anything that makes sense for this.  Building the resolver is essentially the first thing we do, so it seems like its configuration can only come from the user.  But if that is the case, we have (at least?) three options:

1. A different dial option for every `resolver.BuilderOption`
1. Document the options the user shouldn't set (i.e. we will ignore/override the user's setting)
1. `resolver.UserBuilderOptions` as a sub-field of `BuilderOption`  (As an aside, it's bugging me that this wasn't "BuilderOption**s**".  It will break people using it, but...should we change this now before even more people are using it?)

I think I would prefer 2, 3, then 1, in that order.

File: resolver/resolver.go (hunk 1 above)
Comment (lyuxuan) [reply resolver/resolver.go]: done

File: clientconn.go [hunk 3]
```diff
@@ -412,6 +413,16 @@ func WithChannelzParentID(id int64) DialOption {
 	}
 }
 
+// WithServiceConfigOff returns a DialOption that specifies whether resolver
+// should fetch service config. Note that if the serviceConfigOff is set to true,
+// but resolver doesn't honor the setting and still fetch the service config and
+// returns it, grpc will ignore the config.
+func WithServiceConfigOff(a bool) DialOption {
```
Comment (dfawley) [inline clientconn.go]: Nit: could you please change this to "DisableServiceConfig" (throughout)?

File: clientconn.go [hunk 4]
```diff
@@ -412,6 +413,16 @@ func WithChannelzParentID(id int64) DialOption {
 	}
 }
 
+// WithDisableServiceConfig returns a DialOption that specifies whether resolver
+// should fetch service config. Note that if the disableServiceConfig is set to true,
+// but resolver doesn't honor the setting and still fetch the service config and
+// returns it, grpc will ignore the config.
+func WithDisableServiceConfig(a bool) DialOption {
```
Comment (dfawley) [inline clientconn.go]: Are there good reasons to accept a bool here?  If not, I'd prefer to leave it out and make it always disable it (like how `WithBlock` always makes it a blocking operation).

File: clientconn.go [hunk 5]
```diff
@@ -412,6 +413,16 @@ func WithChannelzParentID(id int64) DialOption {
 	}
 }
 
+// WithDisableServiceConfig returns a DialOption that specifies whether resolver
+// should fetch service config. Note that if the disableServiceConfig is set to true,
```
Comment (dfawley) [inline clientconn.go]: Rewording suggestion:

```go
// WithDisableServiceConfig returns a DialOption that causes grpc to ignore any
// service config provided by the resolver and provides a hint to the resolver
// to not fetch service configs.
```

**Pull request merged**